transports/dns/src/lib.rs

```diff
@@ -197,14 +197,11 @@ where
         BoxFuture<'static, Result<Self::Output, Self::Error>>,
     >;
 
-    fn listen_on(
-        &mut self,
-        id: ListenerId,
-        addr: Multiaddr,
-    ) -> Result<(), TransportError<Self::Error>> {
+    fn listen_on(&mut self, addr: Multiaddr) -> Result<ListenerId, TransportError<Self::Error>> {
         self.inner
             .lock()
-            .listen_on(id, addr)
+            .listen_on(addr)
+            .map(ListenerId::map_type::<Self>)
```

Comment:
Interesting. I did not anticipate the need for this. Do you mind explaining this in more detail?

Reply:
Resolved in a different thread.